elasticsearch/src/main/java/io/micronaut/elasticsearch/DefaultElasticsearchClientFactory.java

```diff
@@ -54,6 +63,18 @@ RestClient restClient(DefaultElasticsearchConfigurationProperties elasticsearchC
         return restClientBuilder(elasticsearchConfiguration).build();
     }
 
+    /**
+     * @param elasticsearchConfiguration The {@link DefaultElasticsearchConfigurationProperties} object
+     * @return The Elasticsearch Client
+     */
+    @Singleton
+    ElasticsearchClient elasticsearchClient(DefaultElasticsearchConfigurationProperties elasticsearchConfiguration, ObjectMapper objectMapper) {
+      RestClient restClient = restClientBuilder(elasticsearchConfiguration).build();
+
+      ElasticsearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper(objectMapper));
+      return new ElasticsearchClient(transport);
+    }
+
```

Comment:
If you want you can provide a `ElasticsearchAsyncClient` singleton too - it would be constructed in exactly the same way as the `ElasticsearchClient`.

Reply:
Good idea. I'll add this.

Reply:
If you add this you will probably want to share the transport between both, hence make it a bean and shut it down correctly

Reply:
done